spectator/id.py

```diff
@@ -1,13 +1,25 @@
+import logging
 from typing import Dict, Optional
 
 
 class MeterId:
     def __init__(self, name: str, tags: Optional[Dict[str, str]] = None) -> None:
+        self._logger = logging.getLogger("spectator.MeterId")
+        self.name = name
+
         if tags is None:
             self._tags = {}
         else:
-            self._tags = tags
-        self.name = name
+            self._tags = self._validate_tags(tags)
+
+    def _validate_tags(self, tags: Dict[str, str]) -> Dict[str, str]:
+        valid_tags = {}
+        for k, v in tags.items():
+            if not isinstance(k, str) or not isinstance(v, str):
+                self._logger.warning("%s,%s=%s dropping invalid tag (either key or value is not a str)", self.name, k, v)
```

Comment:
When failing validation, we typically drop the id entirely rather than just drop a specific tag. Ideally this behavior would be consistent. With that behavior, we could also update `spectator.measurements` so the user could detect it in the same way as backend validation errors.

Reply:
Good point. We'll have to think about how that would work in the context of the Registry helper methods, since we don't maintain a central set of MeterIds and users could decide to hold a reference to the returned meter type.

Reply:
Ok, I think I have a reasonable solution for this implemented. This change results in earlier and more verbose feedback to users about what part of the meter is failing validation, while incrementing `spectator.measurements`.

Take another look?